src/top_level_parser.rs

```diff
@@ -25,7 +31,49 @@ enum TopLevelStat {
     Memused(u64),
     MemusedMax(u64),
     LnetMemused(u64),
-    HealthCheck(String),
+    HealthCheck(HealthCheckStat),
+}
+
+fn target_health<I>() -> impl Parser<I, Output = Target>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        string("device"),
+        space(),
+        target(),
+        space(),
+        string("reported"),
+        space(),
+        words(),
```

Comment:
Use some `.skip(..)` here to reduce the size of the tuple needing destructuring.